packages/jest-editor-support/src/Snapshot.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+'use strict';
+
+import traverse from 'babel-traverse';
+import {getASTfor} from './parsers/babylon_parser';
+import {utils} from 'jest-snapshot';
+
+type Node = any;
+
+type SnapshotMetadata = {
+  exists: true | false,
+  name: string,
+  node: Node,
+  content?: string,
+};
+
+const describeVariants = Object.assign(
+  (Object.create(null): {[string]: boolean}),
+  {
+    describe: true,
```

Comment:
I feel like it would be good to put all this in a common package with eslint-plugin-jest, because we are duplicating all of this in multiple places.